### PR TITLE
Docker compose setup

### PR DESCRIPTION
**Pull request by cableman**
This is for internal review and should **_not_** be merged.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -42,5 +46,10 @@ DATAWELL_VENDOR_PROFILE='opac'
 ###< custom ###
 
 ###> enqueue/redis ###
-ENQUEUE_DSN=redis:
+ENQUEUE_DSN=redis://redis:6379
 ###< enqueue/redis ###
+
+###> ES ###
+ESASTICA_HOST=elasticsearch
```
Comment (turegjorup) [inline .env]: `ESASTICA_HOST` -> `ELASTIC_HOST`?

File: .env [hunk 2]
```diff
@@ -42,5 +46,10 @@ DATAWELL_VENDOR_PROFILE='opac'
...
+ESASTICA_PORT=9200
```
Comment (turegjorup) [inline .env]: `ESASTICA_PORT` - `ELASTIC_PORT`?

File: README.md [hunk 3]
```diff
@@ -172,6 +172,32 @@ contain the import logic for the vendors specific access setup
 
 ## Development Setup
 
+### Docker compose
+The project comes with a docker-compose setup base on development only images, that comes with all required extension
```
Comment (turegjorup) [inline README.md]: extension -> extensions

File: config/cloudinary.yml [hunk 4]
```diff
@@ -12,6 +12,6 @@ default:
   transformation: 't_ddb_cover'
   extension: 'jpg'
 
-auto:
-  transformation: 't_ddb_cover,q_auto'
-  extension: 'jpg'
+#auto:
+#  transformation: 't_ddb_cover,q_auto'
+#  extension: 'jpg'
```
Comment (turegjorup) [inline config/cloudinary.yml]: Why? And is there a better way? I assume this has to be undone for prod?

File: config/packages/fos_elastica.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # Read the documentation: https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/Resources/doc/setup.md
 fos_elastica:
     clients:
-        default: { host: localhost, port: 9200 }
+        default: { host: '%env(ESASTICA_HOST)%', port: '%env(ESASTICA_PORT)%' }
```
Comment (turegjorup) [inline config/packages/fos_elastica.yaml]: `ESASTICA_` -> `ELASTIC_`?

File: config/packages/monolog.yml [hunk 6]
```diff
@@ -7,7 +7,7 @@ monolog:
     statistics:
       type: elasticsearch
       elasticsearch:
-        host: localhost
-        port: 9200
+        host: '%env(ESASTICA_HOST)%'
+        port: '%env(ESASTICA_PORT)%'
```
Comment (turegjorup) [inline config/packages/monolog.yml]: ESASTICA_ -> ELASTIC_?

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    volumes:
+      - ${PWD}/.docker/vhost.conf:/etc/nginx/conf.d/default.conf:ro
+      - ./:/app:delegated
+    labels:
+      - "traefik.enable=true"
+      - "traefik.docker.network=frontend"
+      - "traefik.http.routers.${COMPOSE_PROJECT_NAME}.rule=Host(`${COMPOSE_DOMAIN}`)"
+
+  redis:
+    image: 'redis:4'
```
Comment (turegjorup) [inline docker-compose.yml:61]: "Redis 5 was released as GA in October 2018" vs "Redis 4.0 was released as GA in July 2017, newcomers should use Redis 5, but Redis 4 is currently the most production-proven release and will be updated for the next year until Redis 6 will be out."
(https://redis.io/download)

Stay on v4 or go with v5?

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    networks:
+      - app
+    ports:
+      - '6379'
+
+  elasticsearch:
+    image: 'elasticsearch:6.8.4'
```
Comment (turegjorup) [inline docker-compose.yml:68]: 6.8.x is end of life on 2020-11-20. Go with 7.4.x?

**Review by turegjorup — CHANGES_REQUESTED** [review]
Minor comments

**Comment by turegjorup** [comment]
> This is for internal review and should **_not_** be merged.

Consider using ["draft pull request"](https://github.blog/2019-02-14-introducing-draft-pull-requests/) and/or prefixing the title with "WIP" (Work In Progress)

File: .env (hunk 1 above)
Comment (cableman) [reply .env]: The bundle that brings ES to the app is called esatica. But will rename.

File: .env (hunk 2 above)
Comment (cableman) [reply .env]: The bundle that brings ES to the app is called esatica. But will rename.

File: config/cloudinary.yml (hunk 4 above)
Comment (cableman) [reply config/cloudinary.yml]: It has been removed on the current server and I don't remember why, but the links that it creates in the request gives an 404.

Comment (cableman) [reply config/cloudinary.yml]: I think that we concluded that the q_auto was not doing as good a job as expected. Can remember that i was Rolf that can with the auto-quality idea.   The rule have been removed form the CDN.

File: docker-compose.yml (hunk 7 above)
Comment (cableman) [reply docker-compose.yml:61]: We stay to get online now and we don't have time to fight error in redis versions

File: docker-compose.yml (hunk 8 above)
Comment (cableman) [reply docker-compose.yml:68]: We stay as the 7.4.x simply failes the queries and app index create commands. And right now we need to get online fast.

File: .env (hunk 1 above)
Comment (cableman) [reply .env]: Fixed

File: .env (hunk 2 above)
Comment (cableman) [reply .env]: Fixed

File: config/cloudinary.yml (hunk 4 above)
Comment (cableman) [reply config/cloudinary.yml]: Removed it.

File: config/packages/fos_elastica.yaml (hunk 5 above)
Comment (cableman) [reply config/packages/fos_elastica.yaml]: Fixed

File: config/packages/monolog.yml (hunk 6 above)
Comment (cableman) [reply config/packages/monolog.yml]: Fixed

**Review by turegjorup — APPROVED** [review]

**Pull request closed without merging**